plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -408,6 +417,125 @@ public void updatePartitionStatistics(HiveIdentity identity, Table table, String
+            updatePartitionStatisticsBatch(table, keySet);
+        });
+    }
+
+    private void updatePartitionStatisticsBatch(Table table, List<Map.Entry<String, Function<PartitionStatistics, PartitionStatistics>>> updates)
+    {
+        ImmutableList.Builder<BatchUpdatePartitionRequestEntry> updateRequests = ImmutableList.builder();
+        ImmutableSet.Builder<GlueColumnStatisticsProvider.StatisticsUpdate> statisticsUpdates = ImmutableSet.builder();
+
+        Map<String, UpdatePartitionAsyncHandler> asyncHandlers = updates.stream().collect(
+                toImmutableMap(
+                        Entry::getKey,
+                        entry -> new UpdatePartitionAsyncHandler(table, entry.getValue())));
+        Map<String, Future<GetPartitionResult>> getPartitionResults = updates.stream().collect(
+                toImmutableMap(
+                        Entry::getKey,
+                        entry -> glueClient.getPartitionAsync(new GetPartitionRequest()
+                                        .withCatalogId(catalogId)
+                                        .withDatabaseName(table.getDatabaseName())
+                                        .withTableName(table.getTableName())
+                                        .withPartitionValues(toPartitionValues(entry.getKey())),
+                                asyncHandlers.get(entry.getKey()))));
+
+        getPartitionResults.forEach((partitionValues, getPartitionResult) -> {
+            getFutureValue(getPartitionResult);
+
+            UpdatePartitionAsyncHandler asyncHandler = asyncHandlers.get(partitionValues);
+            verify(asyncHandler.isFinished());
+
+            if (asyncHandler.getException().isPresent()) {
+                throw new TrinoException(HIVE_METASTORE_ERROR, asyncHandler.getException().get());
+            }
+
+            PartitionInput partitionInput = GlueInputConverter.convertPartition(asyncHandler.getPartition());
+            partitionInput.setParameters(asyncHandler.getPartition().getParameters());
+
+            updateRequests.add(new BatchUpdatePartitionRequestEntry()
+                    .withPartitionValueList(asyncHandler.getPartition().getValues())
+                    .withPartitionInput(partitionInput));
+            statisticsUpdates.add(new GlueColumnStatisticsProvider.StatisticsUpdate(asyncHandler.getPartition(), asyncHandler.getUpdatedColumnStatistics()));
+        });
+
+        Future<BatchUpdatePartitionResult> updatePartitionResult = glueClient.batchUpdatePartitionAsync(new BatchUpdatePartitionRequest()
+                .withCatalogId(catalogId)
+                .withDatabaseName(table.getDatabaseName())
+                .withTableName(table.getTableName())
+                .withEntries(updateRequests.build()));
+        try {
+            columnStatisticsProvider.updatePartitionStatistics(statisticsUpdates.build());
+            getFutureValue(updatePartitionResult);
```

Comment:
add a comment that we only call `getFutureValue` after we do `columnStatisticsProvider.updatePartitionStatistics` to allow cocurrent operation of updating partition and column statists for sake of performance.